src/frontend/language_phase.nit

```diff
@@ -0,0 +1,121 @@
+# Basic support of internationalization through the generation of id-to-string tables
+module language_phase
+
+private import annotation
+private import parser_util
+import astbuilder
+import gettext
+
+redef class ToolContext
+	# Main phase of `language`
+	var language_phase: Phase = new LanguagePhase(self, null)
+end
+
+class LanguagePhase
+	super Phase
+
+	redef fun process_annotated_node(nmodule, nat) do
+		if not nat.name == "language" then return
+
+		if not nmodule isa AModuledecl then
+			toolcontext.error(nmodule.location, "Cannot use annotation language on something other than a module")
+			return
+		end
+
+		var domain = nmodule.n_name.n_id.text
+
+		if nat.n_args.length != 1 then
+			toolcontext.error(nat.location, "Error: wrong number of arguments for annotation language, expected 1, got {nat.n_args.length}")
+			return
+		end
+		var arg = nat.n_args[0]
+
+		if not arg isa AStringFormExpr then
+			toolcontext.error(nat.location, "Error: expected string argument for annotation language")
+			return
+		end
+
+		var lang_raw = arg.n_string.text
+		var lang = lang_raw.substring(1, lang_raw.length - 2)
+
+		if lang == "" then
+			toolcontext.error(nat.location, "Error: invalid value for argument of annotation, expecting non-empty string")
+		end
+
+		var foldr = nmodule.location.to_s.split("/")
+		foldr.pop
+		var foldr_path = foldr.join("/").realpath
+		foldr_path += "/languages"
```

Comment:
Yup, I forgot that part, I'd need a fallback plan in this case.

I don't really know where to create the language directory in this case ? NIT_DIR ? Since it's not required, it's not that robust either.

I let it be if there's no location ? I guess if it's a virtual module, it would not mind not having a translation since it's temporary ?